CHANGELOG.md

```diff
@@ -1,3 +1,274 @@
+- Check LSP renaming support before prompting for rename text (#6257)
+- Separate diagnostic picker message and code (#6095)
+- Add a config option to bypass undercurl detection (#6253)
+- Only complete appropriate arguments for typed commands (#5966)
+- Discard outdated LSP diagnostics (3c9d5d0)
+- Discard outdated LSP workspace edits (b6a4927)
+- Run shell commands asynchronously (#6373)
+- Show diagnostic codes in LSP diagnostic messages (#6378)
+
+Fixes:
+
+- Fix behavior of `auto-completion` flag for completion-on-trigger (#5042)
+- Reset editor mode when changing buffers (#5072)
+- Respect scrolloff settings in mouse movements (#5255)
+- Avoid trailing `s` when only one file is opened (#5189)
+- Fix erroneous indent between closers of auto-pairs (#5330)
+- Expand `~` when parsing file paths in `:open` (#5329)
+- Fix theme inheritance for default themes (#5218)
+- Fix `extend_line` with a count when the current line(s) are selected (#5288)
+- Prompt: Fix autocompletion for paths containing periods (#5175)
+- Skip serializing JSONRPC params if params is null (#5471)
+- Fix interaction with the `xclip` clipboard provider (#5426)
+- Fix undo/redo execution from the command palette (#5294)
+- Fix highlighting of non-bar cursors (#5575)
+- Fix panic when nooping in `join_selections` and `join_selections_space` (#5423)
+- Fix selecting a changed file in global search (#5639)
+- Fix initial syntax highlight layer sort order (#5196)
+- Fix UTF-8 length handling for shellwords (#5738)
+- Remove C-j and C-k bindings from the completion menu (#5070)
+- Always commit to history when pasting (#5790)
+- Properly handle LSP position encoding (#5711)
+- Fix infinite loop in `copy_selection_on_prev_line` (#5888)
+- Fix completion popup positioning (#5842)
+- Fix a panic when uncommenting a line with only a comment token (#5933)
+- Fix panic in `goto_window_center` at EOF (#5987)
+- Ignore invalid file URIs sent by a language server (#6000)
+- Decode LSP URIs for the workspace diagnostics picker (#6016)
+- Fix incorrect usages of `tab_width` with `indent_width` (#5918)
+- DAP: Send Disconnect if the Terminated event is received (#5532)
+- DAP: Validate key and index exist when requesting variables (#5628)
+- Fix indent guide rendering (#6136)
+- Fix division by zero panic (#6155)
+- Fix scrolloff at view bottom (#6142)
+- Fix lacking space panic (#6109)
+- Send error replies for malformed and unhandled LSP requests (#6058)
+- Fix table column calculations for dynamic pickers (#5920)
+- Skip adding jumplist entries for `:<n>` line number previews (#5751)
+- Fix completion race conditions (#6173)
+- Fix `shrink_selection` with multiple cursors (#6093)
+- Fix indentation calculation for lines with mixed tabs/spaces (#6278)
+- No-op `client/registerCapability` LSP requests (#6258)
+- Send the STOP signal to all processes in the process group (#3546)
+- Fix workspace edit client capabilities declaration (7bf168d)
```

Comment:
Thanks for going trough the effort of splitting changelog messages for my slightly kitchen sinkish PRs. It made sense to bundle these changes from a review perspective but probably makes generating the changelog a bit more annoying